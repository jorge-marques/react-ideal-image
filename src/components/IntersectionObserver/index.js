import React from 'react'
import Observer from 'react-intersection-observer'
import PropTypes from 'prop-types'

const IntersectionObserver = ({onEnter, onLeave, children}) => (
  <Observer onChange={inView => (inView ? onEnter() : onLeave())}>
    {children}
  </Observer>
)

IntersectionObserver.prototype.propTypes = {
  onEnter: PropTypes.func.isRequired,
  onLeave: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
}

export default IntersectionObserver
