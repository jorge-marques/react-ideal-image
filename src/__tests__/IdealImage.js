import React from 'react'
import Waypoint from 'react-waypoint'
import renderer from 'react-test-renderer'
import IdealImage from '../components/IdealImage'
import IntersectionObserver from '../components/IntersectionObserver'

describe('IdealImage', () => {
  it('Renders a snapshot with Waypoint', () => {
    const comp = renderer
      .create(
        <IdealImage
          icon="icon-file"
          icons={{}}
          theme={{}}
          placeholder={{color: 'blue'}}
          srcSet={[
            {
              src: 'some-src.jpg',
              width: 3500,
            },
          ]}
          alt="doggo"
          width={3500}
          height={2095}
          observer={Waypoint}
        />,
      )
      .toJSON()

    expect(comp).toMatchSnapshot()
  })

  it('Renders a snapshot with IntersectionObserver', () => {
    const comp = renderer
      .create(
        <IdealImage
          icon="icon-file"
          icons={{}}
          theme={{}}
          placeholder={{color: 'blue'}}
          srcSet={[
            {
              src: 'some-src.jpg',
              width: 3500,
            },
          ]}
          alt="doggo"
          width={3500}
          height={2095}
          observer={IntersectionObserver}
        />,
      )
      .toJSON()

    expect(comp).toMatchSnapshot()
  })
})
