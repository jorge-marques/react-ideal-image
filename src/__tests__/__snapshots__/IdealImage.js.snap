// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdealImage Renders a snapshot with IntersectionObserver 1`] = `
<observer
  onEnter={[Function]}
  onLeave={[Function]}
>
  <div
    className={undefined}
    onClick={[Function]}
    onKeyPress={[Function]}
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  >
    <svg
      className={undefined}
      height={2095}
      style={undefined}
      width={3500}
    />
  </div>
</observer>
`;

exports[`IdealImage Renders a snapshot with Waypoint 1`] = `
<observer
  onEnter={[Function]}
  onLeave={[Function]}
>
  <div
    className={undefined}
    onClick={[Function]}
    onKeyPress={[Function]}
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  >
    <svg
      className={undefined}
      height={2095}
      style={undefined}
      width={3500}
    />
  </div>
</observer>
`;
